osquery/events/windows/etw/etw_provider_config.h

```diff
@@ -0,0 +1,147 @@
+/**
+ * Copyright (c) 2014-present, The osquery authors
+ *
+ * This source code is licensed as defined by the LICENSE file found in the
+ * root directory of this source tree.
+ *
+ * SPDX-License-Identifier: (Apache-2.0 OR GPL-2.0-only)
+ */
+
+#pragma once
+
+#include <boost/optional.hpp>
+
+#include <osquery/events/windows/etw/etw_data_event.h>
+
+namespace osquery {
+
+class Status;
```

Comment:
What's this for?

Reply:
The `class Status;` sentence here is the forward type declaration to `isValid(`) requires. This is mainly to avoid including the Status header file. The code will not compile without this line.

Reply:
> This is mainly to avoid including the Status header file.

Just curious, why are you trying to avoid including the status header file? I feel like if you're leveraging the `Status` class throughout, you'd want to have it listed in the header includes for readability no? What issue are you avoiding?